### PR TITLE
adds mdatagen as a tool/adopts into splunk otel distribution

### PR DESCRIPTION
**Pull request by hughesjj-splunk**
If this looks good I can add the `//go:generate mdatagen metadata.yaml` to more receivers.

Open to changes on the naming of the make target, feel free to ping me on slack or dm me here.

### CODE REVIEW

**Comment by jeffreyc-splunk** [comment]
I think we'll need to add a `make install-tools` step before this step: https://github.com/signalfx/splunk-otel-collector/blob/main/.github/workflows/integration-test.yml#L48

File: Makefile [hunk 1]
```diff
@@ -115,6 +115,11 @@ gendependabot:
 		  echo "  - package-ecosystem: \"docker\"\n    directory: \"$${dir:1}\"\n    schedule:\n      interval: \"daily\"" >> ${DEPENDABOT_PATH} ); \
 	done
 
+.PHONY: generate
```
Comment (jeffreyc-splunk) [inline Makefile:118]: Does this target need to be explicitly called during CI?

File: Makefile [hunk 2]
```diff
@@ -115,6 +115,11 @@ gendependabot:
...
+generate:
```
Comment (jeffreyc-splunk) [inline Makefile:119]: Does this make sense?
```suggestion
generate: install-tools
```

Comment (hughesjj-splunk) [reply Makefile:119]: oh duh that's how to do it, thank you!

File: Makefile (hunk 1 above)
Comment (hughesjj-splunk) [reply Makefile:118]: Actually, no.  I'd wager you might have some good thoughts on another Makefile etc to place this in.


that said I could see it advantageous as a "wait you actually reran generate and there's no delta from the commit" check if you think it's wise to add such

**Comment by jeffreyc-splunk** [comment]
Need to a `make install-tools` step before https://github.com/signalfx/splunk-otel-collector/blob/main/.github/workflows/integration-test.yml#L48

**Comment by jeffreyc-splunk** [comment]
Do we need to add a `make generate` step before building the collector binaries (https://github.com/signalfx/splunk-otel-collector/blob/main/.github/workflows/build-and-test.yml#L222)?

**Comment by rmfitzpatrick** [comment]
> Do we need to add a `make generate` step before building the collector binaries (https://github.com/signalfx/splunk-otel-collector/blob/main/.github/workflows/build-and-test.yml#L222)?

At least at this initial stage I think generation should be done manually (read: intentionally like the rest of source changes). Immediate implications to auto-generating would be fieldalignment and lint issues that need to modify the commit contents.

File: Makefile [hunk 3]
```diff
@@ -115,6 +115,11 @@ gendependabot:
 		  echo "  - package-ecosystem: \"docker\"\n    directory: \"$${dir:1}\"\n    schedule:\n      interval: \"daily\"" >> ${DEPENDABOT_PATH} ); \
 	done
 
+.PHONY: generate
+generate: install-tools
```
Comment (rmfitzpatrick) [inline Makefile:119]: something like
```suggestion
generated-metrics: install-tools
```
?

File: .github/workflows/integration-test.yml [hunk 4]
```diff
@@ -45,6 +45,7 @@ jobs:
             /home/runner/go/pkg/mod
             /home/runner/.cache/go-build
           key: v1-go-pkg-mod-${{ runner.os }}-${{ hashFiles('**/go.mod', '**/go.sum') }}
+      - run: make install-tools
```
Comment (rmfitzpatrick) [inline .github/workflows/integration-test.yml:48]: Why would this be necessary for integration tests?

**Comment by rmfitzpatrick** [comment]
The internal/tools system works w/ side-effect imports so the mdatagen cmd would need to be added: https://github.com/signalfx/splunk-otel-collector/blob/main/internal/tools/tools.go#L21

**Review by atoulme — APPROVED** [review]

**Comment by jeffreyc-splunk** [comment]
Should we add a new github job to run `make generate-metrics`, and fail if there are uncommitted changes, similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/build-and-test.yml#L184?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**